Lib/typing.py

```diff
@@ -2018,7 +2018,7 @@ def _proto_hook(cls, other):
                 break
 
             # ...or in annotations, if it is a sub-protocol.
-            annotations = getattr(base, '__annotations__', {})
+            annotations = _lazy_annotationlib.get_annotations(base, format=_lazy_annotationlib.Format.FORWARDREF)
             if (isinstance(annotations, collections.abc.Mapping) and
                     attr in annotations and
                     issubclass(other, Generic) and getattr(other, '_is_protocol', False)):
```

Comment:
I think this change means we no longer need to check whether `annotations` is a `Mapping` here, since `get_annotations` is guaranteed to always return a dictionary

```suggestion
            if (attr in annotations
                and issubclass(other, Generic)
                and getattr(other, '_is_protocol', False)):
```